code/game/objects/items/weapons/lighters.dm

```diff
@@ -329,6 +390,41 @@
 	origin_tech = null
 	lit = TRUE
 	w_class = WEIGHT_CLASS_BULKY //to prevent it going to pockets
+	
+/obj/item/match/unathi/cigarette_lighter_act(mob/living/target, mob/living/user, obj/item/direct_attackby_item)
+	var/obj/item/clothing/mask/cigarette/cig = ..()
+	if(!cig)
+		return
```

Comment:
Does this work properly?

Reply:
Yes, it still functions as before, including denying non-fancy lighters the ability to light cigars.